### PR TITLE
Improve build instructions

### PR DESCRIPTION
**Pull request by johnnovak**
Nothing controversial I hope 😄

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -236,9 +246,9 @@ sudo dnf install gcovr lcov
 Run tests and generate report:
 
 ``` shell
-meson setup -Dbuildtype=debug -Db_coverage=true build
-meson test -C build
-meson compile -C build coverage-html
+meson setup -Dbuildtype=debug -Db_coverage=true build/debug
+meson test -C build/debug
+meson compile -C build/debug coverage-html
```
Comment (johnnovak) [inline BUILD.md:251]: This doesn't work @kcgen; could you perhaps update it to the correct invocation?

Comment (kcgen) [reply BUILD.md:251]: Needs either the Clang `lcov` package (which includes both the lcov binary and GenHTML binary), or the GNU `gcovr` package, which can do html generation itself.

```shell
$ meson setup <other flags> -Db_coverage=true build/cov
$ meson compile -C build/cov
$ meson test -C build/cov
$ meson compile coverage-html (or coverage-xml) -C build/cov
```

https://mesonbuild.com/howtox.html#producing-a-coverage-report

Also testing on my side.

Comment (kcgen) [reply BUILD.md:251]: Ok, working on my side.

File: BUILD.md [hunk 2]
```diff
@@ -14,6 +14,16 @@ up builds. The minimum set of dependencies is:
...
 
+## General notes
+
+Meson allows maintaining different build configurations for a number of build
+targets in their own separate subdirectories. E.g. you can keep you debug and
+release configurations in the `build/debug` and `build/release` directories,
+respectively.
+
+If you want to reconfigure an existing target with `meson setup <build_path>`,
+make sure to delete the directory `<build_path>` first.
```
Comment (GranMinigun) [inline BUILD.md:25]: There's no need to do that. First, Meson tries its best to clean up build directory as necessary, so most of the time e.g. switching from debug to release configuration, flipping flags, and even jumping between tags and branches won't cause any problems (in fact, that's the way I usually operate). Another option is to reconfigure, then call `meson setup <build_path> --wipe` to set the build directory up from scratch, but keeping the configuration.

Comment (johnnovak) [reply BUILD.md:25]: Ah, thanks. Somehow I was under the impression that you had to do this. Removing this whole sentence altogether then.

File: BUILD.md (hunk 1 above)
Comment (johnnovak) [reply BUILD.md:251]: Cheers, adding.

File: BUILD.md [hunk 3]
```diff
@@ -14,6 +14,13 @@ up builds. The minimum set of dependencies is:
 All other dependencies are optional and can be disabled while configuring the
 build (in `meson setup` step).
 
+## General notes
+
+Meson allows maintaining different build configurations for a number of build
+targets in their own separate subdirectories. E.g. you can keep your debug and
```
Comment (kcgen) [inline BUILD.md:20]: Maybe a bit simpler: `It's possible to have several Meson build configurations using  subdirectories, for example, you can ..`


`Meson  subdirectories.`

File: docs/build-macos.md [hunk 4]
```diff
@@ -28,8 +32,11 @@ to be installed and the license agreed to:
 2. Install dependencies and related tools:
 
     ``` shell
-    brew install ccache cmake libpng meson libslirp \
-                 opusfile pkg-config python3 sdl2 sdl2_net speexdsp
+    brew install cmake \
+                 pkg-config python3 
+
+    brew install cmake ccache meson libpng sdl2 sdl2_image sdl2_net opusfile \
+         fluid-synth libslirp pkg-config python3 speexdsp
```
Comment (kcgen) [inline docs/build-macos.md:39]: Maybe duplicate on line 35?

**Review by kcgen — COMMENTED** [review]
Just one suggestion and a question. 
Overall looks great, and a big help for those jumping into the code (it's quite a treat having these separate build area!).

File: BUILD.md (hunk 3 above)
Comment (johnnovak) [reply BUILD.md]: Thanks, that reads a lot better :)

File: docs/build-macos.md (hunk 4 above)
Comment (johnnovak) [reply docs/build-macos.md:39]: Oops, good catch!

**Comment by johnnovak** [comment]
> Just one suggestion and a question. Overall looks great, and a big help for those jumping into the code (it's quite a treat having these separate build area!).

Cheers, should be good to go now!

**Pull request merged**